text/0000-artifact-hub-integration.md

```diff
@@ -0,0 +1,595 @@
+    alt publisher has claimed repository ownership in ArtifactHub.io
+    activate publisher
+    publisher->>registry: push buildpack artifact
+    publisher->>repo: update ArtifactHub.io metadata
+    deactivate publisher
+    end
+
+    note over repo, ahindexer: some time elapses...
+
+    activate ahindexer
+    ahindexer->>repo: scan repo
+    repo-->>ahindexer: metadata
+    ahindexer->>ahapi: Write metadata
+    deactivate ahindexer
+
+    note over ahapi, cnbindexer: some time elapses...
+    
+    activate cnbindexer
+    cnbindexer->>ahapi: query api
+    ahapi-->>cnbindexer: metadata
+    cnbindexer->>cnbapi: write metadata
+    deactivate cnbindexer
+    
+    note over cnbapi, dev: some time elapses...
+    
+    alt pack registry-schema=legacy
+    activate dev
+    dev->>cnbapi: pack build, inspect, etc
+    dev->>app: app built
+    deactivate dev
+    
+    else pack registry-schema=artifacthub
+    activate dev
+    dev->>ahapi: pack build, inspect, etc
+    dev->>app: app built
+    deactivate dev
+    end
+```
+
+### Requirement 7 - Terminate the buildpacks registry service
+
+From month 0-6, there will be only additive changes to the registry service. It will maintain full backwards
+compatibility for 6 months.
+
+From month 6-18, the registry service will be read-only. No buildpacks will be able to publish to the registry service
+after 6 months, but it will still serve `GET` requests so that `pack` does not break.
+
+After month 18, the registry service will be terminated. At this point, ArtifactHub.io will be the official source for
+buildpacks.
+
+#### Processes - CNB Registry is terminated
+```mermaid
+sequenceDiagram
+    participant publisher as Buildpack <br/> Publisher
+    participant registry as 3rd Party <br/> Image Registry <br/> (i.e. Docker Hub)
+    participant repo as Publisher Owned <br /> Git Repository
+    participant ahindexer as Artifacthub.io <br/> Indexer
+    participant ahapi as Artifacthub.io <br/> API
+    participant dev as Developer
+    participant app as CNB App
+
+    activate publisher
+    publisher->>registry: push buildpack artifact
+    publisher->>repo: update ArtifactHub.io metadata
+    deactivate publisher
+
+    note over repo, ahindexer: some time elapses...
+
+    activate ahindexer
+    ahindexer->>repo: scan repo
+    repo-->>ahindexer: metadata
+    ahindexer->>ahapi: Write metadata
+    deactivate ahindexer
+
+    note over ahapi, dev: some time elapses...
+    
+    activate dev
+    dev->>ahapi: pack build, inspect, etc
+    dev->>app: app built
+    deactivate dev
+```
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+This is a disruptive change for buildpack publishers. The ownership and publishing model will be different.
+
+This is a destructive change that will affect `pack`. Any versions of `pack` that depend on the buildpack registry will
+break as the various requirements of this RFC are implemented.
+
+ArtifactHub.io will be on the hot path for some `pack` operations.
```

Comment:
What is the uptime like for this service? Does the CNCF have any guarantees about service availability?

What happens if there is an issue with the service? Do we have someone to page? Or a way to open a ticket and get things addressed?

Reply:
Great questions. I will try to get some answers.

Reply:
There is not a published uptime report, nor are there guarantees about service availability. 

There is a #artifact-hub channel in the CNCF Slack and [github issues](https://github.com/artifacthub/hub/issues) that can be used to communicate with maintainers.

Do note that Artifact Hub does not host the packages.